docs/troubleshooting.md

```diff
@@ -46,6 +48,41 @@ This is typically encountered when you attempt to install BKPR with a DNS zone (
+
+```
+Annotations:
+  kubernetes.io/ingress.class:                        nginx
+  kubernetes.io/tls-acme:                             true
+```
+
+`cert-manager` watches for Kubernetes ingress resources. When an ingress resource with the `"kubernetes.io/tls-acme": true` annotation is seen for the first time, `cert-manager` tries to issue an [ACME certificate using HTTP-01 challenge](http://docs.cert-manager.io/en/latest/tutorials/acme/http-validation.html). This works by creating a temporary ingress resource name like `cm-acme-http-solver-${pod_id}` and then triggering the HTTP-01 challenge protocol. This requires that Let's Encrypt is able to resolve the DNS name associated with the Kubernetes ingress resource. On a Kubernetes cluster where BKPR is deployed, this requires that External DNS is functioning properly.
+
+During the time while `cert-manager` negotiates the certificate issue with Let's Encrypt, NGNIX controller temporatily uses a self-signed certificate. This situation should be transient and when it lasts for more than a couple of minutes there is a problem that prevents the certificate from being issued. Please note that when using the staging environment for Let's Encrypt the certificates issued by it not be recognized as valid (the signing CA is not a recognized one). 
+
+When everything works as expected, `cert-manager` will eventually install the certificate as seen next (example):
+
+```
+$ kubectl describe certificate example-com
+Events:
+  Type    Reason          Age      From          Message
+  ----    ------          ----     ----          -------
+  Normal  CreateOrder     57m      cert-manager  Created new ACME order, attempting validation...
+  Normal  DomainVerified  55m      cert-manager  Domain "example.com" verified with "http-01" validation
+  Normal  DomainVerified  55m      cert-manager  Domain "www.example.com" verified with "http-01" validation
+  Normal  IssueCert       55m      cert-manager  Issuing certificate...
+  Normal  CertObtained    55m      cert-manager  Obtained certificate from ACME server
+  Normal  CertIssued      55m      cert-manager  Certificate issued successfully
+```
+
+If this is not the case, Let's Encrypt will refuse to issue the certificate. Besides having NGINX serve a self-signed certificate, this is typically signalled because there are temporary Kubernetes ingress resources named like `cm-acme-http-solver-${pod_id}` lingering around for a while. At some point, certificate generation will time-out, these temporary ingress resources will be destroyed and the Kubernetes certificate will be marked accordingly in its events section. Make sure that External DNS is working properly. Let's Encrypt requires that the DNS name (subject) of the X.509 certificate can resolve properly. Please read the section about troubleshooting DNS and External DNS in this document.
```

Comment:
Following on my previous comment, it seems like the symptoms here are "NGINX serve a self-signed certificate" or "temporary Kubernetes ingress resources named like `cm-acme-http-solver-${pod_id}` lingering". So we could start with that, then explain debugging steps to verify the issues, then explain how things "should work" and the steps to resolve.

Reply:
Can we add a link to the DNS/extran DNS troubleshooting section where you refer to it?